src/service/url-replacements-impl.js

```diff
@@ -128,17 +128,19 @@ export class GlobalVariableSource extends VariableSource {
     });
 
     // Returns the canonical URL for this AMP document.
-    this.set('CANONICAL_URL', this.getDocInfoUrl_('canonicalUrl'));
+    this.set('CANONICAL_URL', () => this.getDocInfo_().canonicalUrl);
 
     // Returns the host of the canonical URL for this AMP document.
-    this.set('CANONICAL_HOST', this.getDocInfoUrl_('canonicalUrl', 'host'));
+    this.set('CANONICAL_HOST', () =>
+      parseUrlDeprecated(this.getDocInfo_().canonicalUrl)['host']);
 
     // Returns the hostname of the canonical URL for this AMP document.
-    this.set('CANONICAL_HOSTNAME', this.getDocInfoUrl_('canonicalUrl',
-        'hostname'));
+    this.set('CANONICAL_HOSTNAME', () =>
+      parseUrlDeprecated(this.getDocInfo_().canonicalUrl)['hostname']);
```

Comment:
we should be able to do `.hostname` here right? this is the Location object?

Reply:
+1 and all other location object field.

Reply:
Done.